inabox/tests/setup_infra.go

```diff
@@ -53,8 +54,11 @@ func SetupGlobalInfrastructure(config *InfrastructureConfig) (*InfrastructureHar
 
 	testConfig := deploy.ReadTestConfig(testName, rootPath)
 
+	// Create a long-lived context for the infrastructure lifecycle
+	infraCtx, infraCancel := context.WithCancel(context.Background())
+
 	if testConfig.Environment.IsLocal() {
-		return setupLocalInfrastructure(ctx, config, testName, testConfig)
+		return setupLocalInfrastructure(setupCtx, infraCtx, infraCancel, config, testName, testConfig)
```

Comment:
this is getting a bit crazy. setup, infra, config, testconfig. do we really need all of these? i have no idea what each of these is doing when looking at this. maybe ok but justwant to mke sure theres no consolidation we could do

Reply:
also whats local vs global infra